sda-commons-server-spring-data-mongo/docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,108 @@
+# Migration Guide
+
+## Getting started
+The spring-data-mongo package is available at the following coordinates:
+```groovy
+implementation "org.sdase.commons:sda-commons-server-spring-data-mongo"
+```
+
+Replacing the MorphiaBundle:
+
+```diff
+-  private final MorphiaBundle<AppConfiguration> morphiaBundle =
+-    MorphiaBundle.builder()
+-      .withConfigurationProvider(AppConfiguration::getMongo)
+-      .withEntityScanPackageClass(ConsentConfigurationEntity.class)
+-      .withValidation()
+-      .build();
+
++  private final SpringDataMongoBundle<AppConfiguration> mongoBundle = 
++    SpringDataMongoBundle.builder()
++      .withConfigurationProvider(AppConfiguration::getMongo)
++      .build();
+```
+
+The Datastore is then available with:
+```java
+public MongoOperations getDatastore() {
+  return this.mongoBundle.getMongoOperations();
+}
+```
+
+### Morphia compatibility
+When upgrading a service from 2.x.x as set of converters is provided to stay compatible with a
+database that was initialised with Morphia. To enable these converters, `.withMorphiaCompatibility()`
+can be used when building the SpringDataMongoBundle.
+
+```java
+SpringDataMongoBundle.builder()
+  .withConfigurationProvider(AppConfiguration::getSpringDataMongo)
+  .withMorphiaCompatibility()
+  build();
+```
+
+### Validation
+Automatic JSR validation is no longer provided in v3. If you still want to validate your models
+you can do so manually using the `ValidationFactory`:
+
+```java
+boolean isValid =
+  Validation.buildDefaultValidatorFactory()
+    .getValidator()
+    .validate(myEntity)
+    .isEmpty();
+```
+
+## Queries
```

Comment:
let's add a description how to replace Morphia's Datastore when initializing the application.